Unstable test TestStaleReadNoExtraTSORequest

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run unit test.

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

No error.

### 3. What did you see instead (Required)

```
panic: tso shouldn't be requested [recovered]
	panic: tso shouldn't be requested

goroutine 538314 [running]:
github.com/pingcap/tidb/executor.(*Compiler).Compile.func1()
	executor/compiler.go:54 +0x477
panic({0x4e3b140, 0x5e78500})
	GOROOT/src/runtime/panic.go:884 +0x213
github.com/pingcap/tidb/session.(*session).PrepareTSFuture(0xc0a0aea000, {0x5eada50, 0xc0b17c29f0}, {0x5e88c40, 0xc0b17c2c90}, {0x5537555, 0x6})
	session/session.go:3715 +0x32c
github.com/pingcap/tidb/sessiontxn/isolation.(*baseTxnContextProvider).replaceTxnTsFuture(0xc0a09116f0, {0x5e88c40, 0xc0b17c2c90})
	sessiontxn/isolation/base.go:389 +0xcf
github.com/pingcap/tidb/sessiontxn/isolation.(*baseTxnContextProvider).prepareTxn(0xc0a09116f0)
	sessiontxn/isolation/base.go:354 +0xf5
github.com/pingcap/tidb/sessiontxn/isolation.(*baseTxnContextProvider).AdviseWarmup(0xc0a09116f0)
	sessiontxn/isolation/base.go:414 +0x90
github.com/pingcap/tidb/session.(*txnManager).AdviseWarmup(0x5f1d6f8?)
	session/txnmanager.go:318 +0x3d
github.com/pingcap/tidb/planner.Optimize({0x5eada50, 0xc0b17c29f0}, {0x5f1d6f8, 0xc0a0aea000}, {0x5ebdf20, 0xc0b17bcfc0}, {0x5ee2a50, 0xc0b17c2a80})
	planner/optimize.go:194 +0x7ff
github.com/pingcap/tidb/executor.(*Compiler).Compile(0xc0ae0894b8, {0x5eada50?, 0xc0b17c29f0?}, {0x5ec3380?, 0xc0b17bcfc0})
	executor/compiler.go:98 +0x5cc
github.com/pingcap/tidb/session.(*session).ExecuteStmt(0xc0a0aea000, {0x5eada50?, 0xc0b17c29f0?}, {0x5ec3380?, 0xc0b17bcfc0?})
	session/session.go:2140 +0x62c
github.com/pingcap/tidb/session.(*session).ExecuteInternal(0xc0a0aea000, {0x5eada50, 0xc0b17c29f0}, {0xc08590dee0, 0xcb}, {0x0, 0x0, 0x0})
	session/session.go:1628 +0x268
github.com/pingcap/tidb/ddl.(*session).execute(0xc0a0ab55c0, {0x5ead9e0, 0xc000234000}, {0xc08590dee0, 0xcb}, {0x5558733, 0xf})
	ddl/ddl.go:1639 +0x216
github.com/pingcap/tidb/ddl.(*ddl).getJob(0xc0a08a5600, 0xc0fad0899968fd40?, 0x160242afc1?, 0xc0ae089ae0)
	ddl/job_table.go:113 +0x175
github.com/pingcap/tidb/ddl.(*ddl).getGeneralJob(0xc0ae089b78?, 0x1?)
	ddl/job_table.go:143 +0x4f
github.com/pingcap/tidb/ddl.(*ddl).loadDDLJobAndRun(0xc0a08a5600, 0xc0ae089d30?, 0xc0a18ddf50, 0xc0ae089dc0)
	ddl/job_table.go:225 +0x139
github.com/pingcap/tidb/ddl.(*ddl).startDispatchLoop(0xc0a08a5600)
	ddl/job_table.go:209 +0x7d4
github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()
	util/wait_group_wrapper.go:154 +0x6b
created by github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run
	util/wait_group_wrapper.go:152 +0x8d
--
```

### 4. What is your TiDB version? (Required)

0212c5056a

<!-- Paste the output of SELECT tidb_version() -->

--- a/session/session.go
+++ b/session/session.go
@@ -3711,7 +3711,7 @@ func (s *session) PrepareTSFuture(ctx context.Context, future oracle.Future, sco
 	}
 
 	failpoint.Inject("assertTSONotRequest", func() {
-		if _, ok := future.(sessiontxn.ConstantFuture); !ok {
+		if _, ok := future.(sessiontxn.ConstantFuture); !ok && !s.isInternal() {
 			panic("tso shouldn't be requested")
 		}
 	})

